### PR TITLE
Added mock ups and user functionalities

### PR DESCRIPTION
**Pull request by ghost**
Primary minimal testing done.

### CODE REVIEW

**Comment by in42** [comment]
@tauoverpi add some screenshots

File: app/routes.js
Comment (in42) [inline app/routes.js:6]: I think `import  AuthPage from './auth/components/AuthPage` will work. `index` is not needed.

**Review by in42 — APPROVED** [review]
Seems ok.

File: app/routes.js
Comment (ghost) [reply app/routes.js:6]: Oh this was done by Webstorm when I moved the file. I did not notice.

**Comment by ghost** [comment]
image
image
image
image
image
image
image

File: app/user/profile/ProfilePage.js
Comment (sagnikm95) [inline app/user/profile/ProfilePage.js:95]: @tauoverpi  @in42  The fields are not editable. u need to bind the input fields to a state variable and then define a  on change event handler on those fields in which u set the input contents to the state variable

File: server/routes/commonPostApi.js
Comment (sagnikm95) [inline server/routes/commonPostApi.js:12]: @tauoverpi  @devanshu0987  are we keeping a common post api for sure?? there will be some situations in which the  way the posts are viewed will differ [eg : admin side favourites ; apply will not be there ; there will be some posts which will be displayed only on the admin side ] - do we deal with these on the client side or have a different api for admin and user [ common model ]

File: server/controllers/userPostController.js
Comment (sagnikm95) [inline server/controllers/userPostController.js:38]: if you are keeping a common post api it makes no sense to  name the controllers "userPostControllers';  
One solution using a common post api cd be to keep 3 layers of controller
- commonPost [ get all post etc]
- user specific
- admin specific

File: server/controllers/userPostController.js
Comment (sagnikm95) [inline server/controllers/userPostController.js:21]: Why do you need this?

Comment (devanshu0987) [reply server/controllers/userPostController.js:21]: I think there is no need of this.

**Comment by devanshu0987** [comment]
We should decide on the Post API schematics. I am suggesting that we should write all the API at one place and later employ a higher level middleware to check role and accordingly give access to USER and ADMIN. At this point, its all hotch-potch.

**Comment by sagnikm95** [comment]
@tauoverpi  i agree with @devanshu0987  on the Post API schematics. Decide which one of you will refactor the code.

File: app/user/profile/ProfilePage.js
Comment (ghost) [reply app/user/profile/ProfilePage.js:95]: I know these are not editable. It's just a mock up. I do not know what the fields would be, so I just made something for the page. Discuss it over with Sohail and then we can work on it.

File: server/routes/commonPostApi.js
Comment (ghost) [reply server/routes/commonPostApi.js:12]: Right now we have assigned all the rights of an user to admin  i.e. an admin can do whatever an user can + some more. I know this might not be the best thing, but @in42 and I had a bit of discussion and we decided to go along with that, as that felt best at that moment,

File: server/controllers/userPostController.js
Comment (ghost) [reply server/controllers/userPostController.js:21]: Yes this was from before @in42 merge, so I did not have any `user` things to work with, so used this. But now I have changed and refactored to use from @in42 code. I will remove this in the next commit.

File: server/controllers/userPostController.js
Comment (ghost) [reply server/controllers/userPostController.js:38]: I agree this name is not sensible, right now the api we have is admin rights are superset of user rights, so what I can do is change its name following the `routes/` folder names.

**Comment by ghost** [comment]
Yes I agree too. But this pr does not only have `api` codes, so may be do the whole refactoring thing in the next pr. Let's close this if you are agreeing on other things than the api. Then may be have a discussion and refactor the whole api thing. And as to who would do the refactoring we can discuss over that too. And @sagnikm95 please talk to Sohail and let's have all things finalised once and for all, and then get back to work.

**Comment by sagnikm95** [comment]
Ok. I am merging yours. 
And i had a talk with Sohail. we are going to meet soon and discuss the prototypes

**Pull request merged**
